### DISCUSSION TITLE
firebaseAdminInitConfig object in init function does not require all necessary fields for firebase-admin.

### DISCUSSION THREAD

**Discussion by kevinnammour** [gladly-team/next-firebase-auth#624] [Q&A] [answered]
When setting up firebase-admin, I used to specify all of those attributes:
`import admin from "firebase-admin";

const serviceAccount = {
  type: process.env.FIREBASE_ADMIN_TYPE,
  project_id: process.env.FIREBASE_ADMIN_PROJECT_ID,
  private_key_id: process.env.FIREBASE_ADMIN_PRIVATE_KEY_ID,
  private_key: process.env.FIREBASE_ADMIN_PRIVATE_KEY,
  client_email: process.env.FIREBASE_ADMIN_CLIENT_EMAIL,
  client_id: process.env.FIREBASE_ADMIN_CILENT_ID,
  auth_uri: process.env.FIREBASE_ADMIN_AUTH_URI,
  token_uri: process.env.FIREBASE_ADMIN_TOKEN_URI,
  auth_provider_x509_cert_url:
    process.env.FIREBASE_ADMIN_AUTH_PROVIDER_x509_CERT_URL,
  client_x509_cert_url: process.env.FIREBASE_ADMIN_CLIENT_x509_CERT_URL,
};

if (!admin.apps.length) {
  admin.initializeApp({
    credential: admin.credential.cert(serviceAccount),
    storageBucket: "gs://dormati-b29a7.appspot.com",
  });
}

export { admin };`

However, now, in firebaseAdminInitConfig, only those are required:

`firebaseAdminInitConfig: {
      credential: {
        projectId: process.env.FIREBASE_ADMIN_PROJECT_ID as string,
        clientEmail: process.env.FIREBASE_ADMIN_CLIENT_EMAIL as string,
        privateKey: process.env.FIREBASE_ADMIN_PRIVATE_KEY as string,
      },
      databaseURL: process.env.NEXT_PUBLIC_FIREBASE_DATABASE_URL as string,
    },`

I don't want something to go wrong but how can I specify all the other attributes that you didn't ask for?

**Comment by kmjennison** [comment, marked as answer]
We initialize `firebase-admin` by spreading the properties you define in the `firebaseAdminInitConfig` object, so you can add whatever you need:
https://github.com/gladly-team/next-firebase-auth/blob/v1.x/src/initFirebaseAdminSDK.js#L24
